### DISCUSSION TITLE
Some bug reports for Audio Share

### DISCUSSION THREAD

**Discussion by RobertPeetsalu** [mkckr0/audio-share#64] [Q&A]
First, thanks for the effort of making the app. I hope my feedback will help improve it.

I had no problems installing the android APK v0.0.17 from F-Droid and running the Windows exe with the same version, (although Windows didn't like the "publisher Unknown" and Android got all panicky about installing unknown stuff) but I ran into 2 problems when I tried to play some music:

**Sound _signal_ is quiet and distorted**
The first problem is that the sound played from the phone's audio server is distorted. I made a recording: 
https://github.com/mkckr0/audio-share/assets/892058/7c4a9b6d-151f-407a-afb4-0f03063b6cc2

The only things that I changed in server settings were:
- I set to ignore battery optimizations in hopes that it will fix the problem - it didn't help.
- I set Loudness Enhancer to 1000mB. Otherwise the sound was so quiet that I couldn't hear it over my laptop's built-in speakers. At 1000mB the sound became as loud as normal volume on my laptop speakers. So I assume it's quieter than it should be.

If I play the same music directly from my phone, there are no distortion or volume problems.
I also tried plugging my stereo speakers to the phone and they also played with distortion.

**Can't set the phone as the only speaker for the PC**
I installed the app so that I could plug my sound system to my phone instead of my Laptop, so I wouldn't have to drag cables with me when I move with the laptop. So I need to play PC sound from the phone only.

This leads us to the second problem - audio server on the laptop doesn't create a new audio output option for Windows, but rather repeats sound coming from the existing default sound output, which in my case are "Speakers (Realtek (R) Audio)". As a result when I play music, I hear it from the laptop speakers and with some delay from the phone. 

I looked for a workaround in the volume mixer, but there I could only either mute everything or only mute the audio server as a program. I can't mute the speakers without also muting the audio server, because the speakers work as the source of sound for the server. 

My phone: Xiaomi Redmi Note 9 Pro running MIUI Global 12.0.3 and up to date Android 10.
My laptop: ASUS TUF Gaming A15 with updated Windows 11 and updated Realtek audio drivers.

**Comment by RobertPeetsalu** [comment]
here's my server log
[server.log](https://github.com/mkckr0/audio-share/files/14858618/server.log)

and config:
[App]
AutoRun=1
Running=0
[Network]
host=192.168.3.4
port=65530
[Audio]
endpoint=Speakers (Realtek(R) Audio)

**Comment by mkckr0** [comment]
You can mute the laptop by https://github.com/mkckr0/audio-share/issues/4#issuecomment-1726721684

For the lower volume, it's a known bug but not clear about the reason. "Loudness Enhancer" is a temporary measure. #34

For the distorted sound, it may be caused by the network. You can try to use a USB tethering to test it.

**Reply by RobertPeetsalu** [reply]
Bad news:
1. Muting the laptop as described in that comment also mutes the music from the phone. The only workaround I could think of is making music very quiet on the laptop and amplified in the phone's audio server settings, but that would hurt my ears every time I forgot to turn the volume down in the laptop before connecting. 
2. When I use USB tether to share my phone's internet with the laptop and turn off the laptop's wifi, the tethering itself works (web pages load on the laptop) and audio server on the laptop can be launched, but the audio server on the phone immediately throws a "connection timed out" when trying to start it. Same happens when I try a different host IP from the one that worked over the WiFi connection. So no success trying to use USB tether. 

Good news:
After restsarting my laptop (because volume control went crazy - I couldn't change or mute my volume anymore) and trying to connect over WiFi again, the connection worked and there's no more distortion. Sound is still as quiet as before, but now it is only delayed, not distorted.

So - in case of distortion or sound controls not working, restarting Windows helps, but now I need a way to mute the laptop without muting the phone. For me the short delay is probably bearable if I would only hear the delayed audio from my stereo plugged to the phone, not the original audio from the laptop as well.

**Reply by mkckr0** [reply]
You can install a virtual audio device https://github.com/mkckr0/audio-share#method-2-install-a-third-party-virtual-audio-device-driver

So the laptop won't make any sound.

**Reply by RobertPeetsalu** [reply]
ahh, the distortion is back again with no apparent cause =S

**Reply by RobertPeetsalu** [reply]
Installing Steam Link on the laptop added the Steam Streaming Speakers output and using that as output works - now the laptop sound comes only from the phone.

**Reply by RobertPeetsalu** [reply]
Next problem encountered is - the mobile audio server sometimes has gaps in sound and says that there's a "heartbeat timeout". After googling a bit it seems to have something to do with "applications that perform lengthy processing in the same thread" without adding regular heartbeat messages. Is there some workaround to prevent those?

**Reply by mkckr0** [reply]
It's a network problem that your network quality is low and the delay is over the limit. USB tethering is the best measure because it provides a wired network connection. The "connection timed out" may be because you didn't choose a right IP address for "AudioShareServer.exe"

You can also use the Mobile Hotspot to connect you phone to laptop directly without a WiFi router. It may also be effective.
